### PR TITLE
move engine to sprig repo

### PR DESCRIPTION
**Pull request by recursiveforte**

### CODE REVIEW

**Comment by recursiveforte** [comment]
Related: https://github.com/hackclub/sprig-engine/pull/5

**Comment by Jonak-Adipta-Kalita** [comment]
i couldnt agree more!

File: firmware/spade/src/rpi/upload.h [hunk 1]
```diff
@@ -36,7 +70,7 @@ static struct {
 
 static void upl_flush_buf(void) {
   uint32_t interrupts = save_and_disable_interrupts();
-  flash_range_program(FLASH_TARGET_OFFSET + (upl_state.page++) * 256,
+  flash_range_program(FLASH_TARGET_OFFSET(slot) + (upl_state.page++) * 256,
```
Comment (JosiasAurel) [inline firmware/spade/src/rpi/upload.h:73]: this seems to be modifying the engine code of the firmware

**Review by JosiasAurel — CHANGES_REQUESTED** [review]
we should exclude changes unrelated to the sprig engine. besides that this looks okay to me!

File: firmware/spade/src/rpi/upload.h (hunk 1 above)
Comment (recursiveforte) [reply firmware/spade/src/rpi/upload.h]: oop, this was accidental

**Review by JosiasAurel — APPROVED** [review]
lgtm

**Pull request merged**